File: Tests/executors/CSharp/CoreExecutors/DynamicResources.cs
using System.Text.Json.Nodes;

using Framework;

public class DynamicResources : HeavyOperationStatelessExecutor
{
	public override void DoGet(HttpRequest request, HttpResponse response)
	{
		byte[] fileData = request.GetFile("page.md");

		response.SetBody(request.ProcessStaticFile(fileData, ".md"));
	}

	public override void DoPost(HttpRequest request, HttpResponse response)
	{
		byte[] fileData = request.GetFile("print.wfdp");
		JsonObject arguments = [];

		arguments["@print"] = new JsonObject()
		{
			["data"] = request.GetJson()["data"].GetString()!
		};

		response.SetBody(request.ProcessDynamicFile(fileData, arguments));
	}
}

File: Tests/executors/CXX/core/src/DynamicResources.cpp
#include "DynamicResources.h"

void DynamicResources::doGet(framework::HttpRequest& request, framework::HttpResponse& response)
{
	std::string fileData = request.getFile("page.md");

	response.setBody(request.processStaticFile(fileData, ".md"));
}

void DynamicResources::doPost(framework::HttpRequest& request, framework::HttpResponse& response)
{
	std::string fileData = request.getFile("print.wfdp");
	framework::JsonObject arguments;

	arguments["@print"] = request.getJson().getParsedData();

	response.setBody(request.processDynamicFile(fileData, arguments));
}

DEFINE_EXECUTOR(DynamicResources)
